src/metrics/integration/match_rule_update_stress_test.go

```diff
@@ -222,12 +222,21 @@ func TestMatchWithRuleUpdatesStress(t *testing.T) {
 		wg.Add(2)
 		go func() {
 			defer wg.Done()
+			it := iterPool.Get()
```

Comment:
this stress does a good job demonstrating the change works. it's reusing the same iter for all match requests.